src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx

```diff
@@ -164,17 +152,65 @@ export const BorrowInput = ({
   const useLendingQuoteQueryArgs = useMemo(
     () => ({
       collateralAssetId,
+      collateralAccountId,
+      borrowAccountId,
       borrowAssetId: borrowAsset?.assetId ?? '',
-      depositAmountCryptoPrecision: depositAmount ?? '0',
+      depositAmountCryptoPrecision: cryptoDepositAmount ?? '0',
     }),
-    [borrowAsset?.assetId, collateralAssetId, depositAmount],
+    [
+      borrowAccountId,
+      borrowAsset?.assetId,
+      collateralAccountId,
+      collateralAssetId,
+      cryptoDepositAmount,
+    ],
   )
   const {
     data,
     isLoading: isLendingQuoteLoading,
     isError: isLendingQuoteError,
+    error: lendingQuoteError,
   } = useLendingQuoteOpenQuery(useLendingQuoteQueryArgs)
 
+  const { data: estimatedFeesData, isLoading: isEstimatedFeesDataLoading } =
+    useQuoteEstimatedFeesQuery({
+      collateralAssetId,
+      collateralAccountId,
+      borrowAccountId,
+      borrowAssetId: borrowAsset?.assetId ?? '',
+      depositAmountCryptoPrecision: cryptoDepositAmount ?? '0',
+    })
+
+  const hasEnoughBalance = useMemo(
+    () =>
+      bnOrZero(cryptoDepositAmount)
+        .plus(
+          bnOrZero(estimatedFeesData?.txFeeCryptoBaseUnit).div(
+            bn(10).pow(collateralAsset?.precision ?? '0'),
+          ),
+        )
+        .lte(amountAvailableCryptoPrecision),
+    [
+      amountAvailableCryptoPrecision,
+      collateralAsset?.precision,
+      cryptoDepositAmount,
+      estimatedFeesData?.txFeeCryptoBaseUnit,
+    ],
+  )
+
+  const quoteErrorTranslation = useMemo(() => {
+    if (!hasEnoughBalance) return 'common.insufficientFunds'
+    if (isLendingQuoteError) {
+      if (
+        /not enough fee/.test(lendingQuoteError.message) ||
+        /not enough to pay transaction fee/.test(lendingQuoteError.message)
```

Comment:
could be worth ignoring case
```suggestion
        /not enough fee/i.test(lendingQuoteError.message) ||
        /not enough to pay transaction fee/i.test(lendingQuoteError.message)
```

Reply:
why not 🧠 d9507a34067cdb1221f5228a17e6b4b704a2ab9c